src/machine/machine_atsamd51.go

```diff
@@ -1733,36 +1734,81 @@ func (pwm PWM) getMux() PinMode {
 
 // USBCDC is the USB CDC aka serial over USB interface on the SAMD21.
 type USBCDC struct {
-	Buffer *RingBuffer
+	Buffer  *RingBuffer
+	TxIdx   volatile.Register8
+	waitTxc bool
+	sent    bool
 }
 
-// WriteByte writes a byte of data to the USB CDC interface.
-func (usbcdc USBCDC) WriteByte(c byte) error {
-	// Supposedly to handle problem with Windows USB serial ports?
+// Flush flushes buffered data.
+func (usbcdc *USBCDC) Flush() error {
 	if usbLineInfo.lineState > 0 {
-		// set the data
-		udd_ep_in_cache_buffer[usb_CDC_ENDPOINT_IN][0] = c
+		idx := usbcdc.TxIdx.Get()
+		sz := idx & 0x3F
```

Comment:
I suggest replace `0x3F` with some const like `maxBufferIdx` to make it more clear instead of the magic value.

Reply:
I named it usbcdcTxSizeMask.